Documentation/pic_programmer.md

```diff
@@ -0,0 +1,277 @@
+    Setup peripherals...
+    (OK)
+    Found '18F24J50' in programming database :) index = 1
+    Checking for 18F24J50 attached to programmer...
+    Found 18F24J50 (0X4C02, ID: 0X260 REV: 0X2)
+    Reading HEX file 'OLSv1-bootloader-v1-20MHz.hex' ... Read binary size = 16384
+    Erasing chip... OK :)
+    Writing page 0, 0000...
+    Writing page 1, 0100...
+    ...
+    Writing page 7, 0700...
+    Writing page 8, 0800...
+    Skipping page 9 [ 0x000900 ], not used
+    ...
+    Writing page 63, 3f00...
+    Reading page 0, 0000...
+    Reading page 1, 0100...
+
+Programming a 24FJ64GA002
+-------------------
+    E:\picprog>picprog -p buspirate -u COM12 -s 115200 -c 24FJ64GA002 -t HEX -w BPv3-Bootloader-v4.4.hex -E -V
+    (Bus) Pirate PIC Programer v0.1
+
+    buspirate
+    Initializing interface
+    115200
+    Entering binary mode
+    BP: Setup mode...
+    Setup peripherals...
+    (OK)
+    Found '24FJ64GA002' in programming database :) index = 2
+    Checking for 24FJ64GA002 attached to programmer...
+    Found 24FJ64GA002 (0X447, ID: 0X447 REV: 0X3043)
+    Reading HEX file 'BPv3-Bootloader-v4.4.hex' ... Read binary size = 88064
+    Erasing chip... OK :)
+    Writing page 0, 0000...
+    Writing page 1, 0020...
+    Writing page 2, 0040...
+
+Programming a 18F2550
+-------------------
+    E:\picprog>picprog -p buspirate -u COM12 -s 115200 -c 18F2550 -E
+    (Bus) Pirate PIC Programer v0.1
+
+    buspirate
+    Initializing interface
+    115200
+    Entering binary mode
+    BP: Setup mode...
+    Setup peripherals...
+    (OK)
+    Found '18F2550' in programming database :) index = 0
+    Checking for 18F2550 attached to programmer...
+    Found 18F2550 (0X1247, ID: 0X92 REV: 0X7)
+    Erasing chip... OK :)
+
+This demo requires the Bus Pirate high-voltage programming adapter or a similar 13volt power supply and control circuit. 18F2550 programming is not yet complete.
+
+picprog Command Line Options
+-------------------
+    picprog -p PROG -u PORT -s SPEED -c CHIP -t TYPE -w | -r FILE -E | -W | -R | -V
+
+  * -p PROG
+    * Name of the interface
+  * -u PORT
+    * Interface port
+  * -s SPEED
+    * Interface speed
+  * -c CHIP
+    * Chip type
+  * -t TYPE
+    * Input/output file format {HEX,BIN}
+  * -w FILE
+    * File to be written to the PIC
+  * -r FILE
+    * File to be read from the PIC
+  * -E
+    * Erase Flash
+  * -V
+    * Verify content of flash
+  * -I
+    * Read the chip ID
+  * -h
+    * Display usage help
+
+Add new PICs
+-------------------
+PiratePICprog is fairly feature complete, we just need to add settings for a bunch of PIC chips. We really need a tutorial that shows how to do this, but for now you can refer to the [existing chips](http://code.google.com/p/dangerous-prototypes-open-hardware/source/browse/trunk/PiratePICprog/pic.c#22). Just send us your settings and we'll compile a new release.
+
+### Source location
+_TODO: Where?_ code.google.com svn is permanently disabled...
```

Comment:
I found a copy here: https://github.com/DangerousPrototypes/PiratePICprog/tree/master/software - I'll probably create a new repo later and clean it up.